components/event_dispatcher/container_aware_dispatcher.rst

```diff
@@ -4,103 +4,7 @@
-only be created if an event is dispatched that requires those listeners.
-
-Setup
------
-
-Setup is straightforward by injecting a :class:`Symfony\\Component\\DependencyInjection\\ContainerInterface`
-into the :class:`Symfony\\Component\\EventDispatcher\\ContainerAwareEventDispatcher`::
-
-    use Symfony\Component\DependencyInjection\ContainerBuilder;
-    use Symfony\Component\EventDispatcher\ContainerAwareEventDispatcher;
-
-    $container = new ContainerBuilder();
-    $dispatcher = new ContainerAwareEventDispatcher($container);
-
-Adding Listeners
-----------------
-
-The ``ContainerAwareEventDispatcher`` can either load specified services
-directly or services that implement :class:`Symfony\\Component\\EventDispatcher\\EventSubscriberInterface`.
-
-The following examples assume the service container has been loaded with
-any services that are mentioned.
-
-.. note::
-
-    Services must be marked as public in the container.
-
-Adding Services
-~~~~~~~~~~~~~~~
-
-To connect existing service definitions, use the
-:method:`Symfony\\Component\\EventDispatcher\\ContainerAwareEventDispatcher::addListenerService`
-method where the ``$callback`` is an array of ``array($serviceId, $methodName)``::
-
-    $dispatcher->addListenerService($eventName, array('foo', 'logListener'));
-
-Adding Subscriber Services
-~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Event subscribers can be added using the
-:method:`Symfony\\Component\\EventDispatcher\\ContainerAwareEventDispatcher::addSubscriberService`
-method where the first argument is the service ID of the subscriber service,
-and the second argument is the service's class name (which must implement
-:class:`Symfony\\Component\\EventDispatcher\\EventSubscriberInterface`) as follows::
-
-    $dispatcher->addSubscriberService(
-        'kernel.store_subscriber',
-        'StoreSubscriber'
-    );
-
-The ``EventSubscriberInterface`` is exactly as you would expect::
-
-    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
-    use Symfony\Component\HttpKernel\KernelEvents;
-    // ...
-
-    class StoreSubscriber implements EventSubscriberInterface
-    {
-        public static function getSubscribedEvents()
-        {
-            return array(
-                KernelEvents::RESPONSE => array(
-                    array('onKernelResponsePre', 10),
-                    array('onKernelResponsePost', 0),
-                ),
-                'store.order'     => array('onStoreOrder', 0),
-            );
-        }
-
-        public function onKernelResponsePre(FilterResponseEvent $event)
-        {
-            // ...
-        }
-
-        public function onKernelResponsePost(FilterResponseEvent $event)
-        {
-            // ...
-        }
-
-        public function onStoreOrder(FilterOrderEvent $event)
-        {
-            // ...
-        }
-    }
+    The ``ContainerAwareEventDispatcher`` was removed in Symfony 4.0. Use
+    ``EventDispatcher`` with closure-proxy injection instead.
```

Comment:
reference article/section about closure-proxy injection?

Reply:
I don't understand this comment. Is there an article that we could link to in this comment? Or are you suggesting to create a new article? Thanks!